core/src/main/java/cz/xtf/core/bm/BuildManager.java

```diff
@@ -15,6 +16,9 @@ public BuildManager(OpenShift openShift) {
 		if (openShift.getProject(openShift.getNamespace()) == null) {
 			openShift.createProjectRequest();
 		}
+		if (OpenShiftConfig.pullSecret() != null) {
```

Comment:
I suspect there may be a race between the project being created and the creation of the secret in the namespace.   ( openShift.createProjectRequest(); is async afaik)

Reply:
Yes, that's true. I was planing to create something like `isProjectReady` waiter.

Reply:
I'll mark this as resolved through `isProjectReady()` addition. Let me know if you don't like it. :)